src/components/Table.js

```diff
@@ -113,15 +113,28 @@ export class Table extends Component {
     }
 
     nextPage() {
-        this.setState(currentState => {
-            return nextPage({ state: currentState })
-        });
+        if (!this.props.updateData) {
+            this.setState(currentState => {
+                return nextPage({ state: currentState })
+            });
+            return 
+        }
+
+        const { currentPage, totalPages } = this.state.pagination;
+        if (currentPage < totalPages) this.props.updateData({page: currentPage + 1});
```

Comment:
What updates currentPage?

Reply:
This calls a function `updateData` that will send a request to the server, and in the server response the `current_page` is sent, and then comes back into here via updated props.